### PR TITLE
Remove migrated CircleCI jobs and cleanup script

### PR DESCRIPTION
**Pull request by eloquence**
That part of the job has been migrated to `securedrop-yum-test` . The remainder will be migrated to `securedrop-workstation`.

The cleanup script has been split into Debian and RPM portions, which are running in `securedrop-apt-test` and `securedrop-yum-test`, respectively. 

Update 2024-04-15: Also removes the build/push jobs migrated via https://github.com/freedomofpress/securedrop-workstation/pull/955

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,25 +103,18 @@ jobs:
...
-            ~/project/scripts/clean-old-packages workstation/dom0/f32-nightlies 7
             git add .
             # If there are changes, diff-index will fail, so we commit
             git diff-index --quiet HEAD || git commit -m "Automated SecureDrop workstation build"
-            # And clean up non-nightly packages too
-            ~/project/scripts/clean-old-packages workstation/dom0/f32 4
-            git add .
-            git diff-index --quiet HEAD || git commit -m "Cleanup old packages"
-
-            git push origin main
```
Comment (legoktm) [inline .circleci/config.yml:124]: This push is still needed, because it's what pushes the nightly commit.

**Review by legoktm — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (eloquence) [reply .circleci/config.yml:124]: Oof, of course, sorry about that & thanks for the catch.

**Comment by eloquence** [comment]
RPM build failure potentially related to 4.2 changes that just landed? Can poke tomorrow.

**Comment by legoktm** [comment]
> RPM build failure potentially related to 4.2 changes that just landed? Can poke tomorrow.

Yep, and that further justifies putting that nightly logic into the workstation repo...probably we can do that and then come back to this cleanup? I did start in https://github.com/freedomofpress/securedrop-workstation/pull/955/, I think it just needs testing.

**Comment by eloquence** [comment]
> Yep, and that further justifies putting that nightly logic into the workstation repo...probably we can do that and then come back to this cleanup? I did start in https://github.com/freedomofpress/securedrop-workstation/pull/955, I think it just needs testing.

OK, since this is directly tied to the 4.2 work, maybe best for you to take that one over the line, and I can take point if review unless someone else grabs it first?

Meanwhile, I can poke at the lint/reprotest jobs in this repo.

**Comment by legoktm** [comment]
> OK, since this is directly tied to the 4.2 work, maybe best for you to take that one over the line, and I can take point if review unless someone else grabs it first?

Sounds good!

**Comment by eloquence** [comment]
Nightlies triggered directly from `securedrop-workstation` appear to be working, so I've also removed the build/push CircleCI config in this PR. Once this PR lands, I believe we can safely remove the SSH key associated with this repository as well, as I don't see it being used for anything else.

**Review by legoktm — APPROVED** [review]
\o/

I will file an infra ticket to remove the SSH key

**Pull request merged**